sys/net/application_layer/nanocoap/nanocoap.c

```diff
@@ -846,27 +846,35 @@ ssize_t coap_opt_add_string(coap_pkt_t *pkt, uint16_t optnum, const char *string
     return write_len;
 }
 
-ssize_t coap_opt_add_uquery(coap_pkt_t *pdu, const char *key, const char *val)
+ssize_t coap_opt_add_uquery(coap_pkt_t *pkt, const char *key, const char *val)
 {
+    return coap_opt_add_uquery2(pkt, key, strlen(key), val, val ? strlen(val) : 0);
+}
+
+ssize_t coap_opt_add_uquery2(coap_pkt_t *pkt, const char *key, size_t key_len,
+                             const char *val, size_t val_len)
+{
+    assert(pkt);
+    assert(key);
+    assert(key_len);
+    assert((!val || !val_len) || (val && val_len));
```

Comment:
This line is not correct. It is always returns true. Should be:

```
assert(!val_len || (val && val_len))
```

Reply:
Maybe we could be *very* explicit here and do `assert ((!val && !val_len) || (val && val_len))` to clearly state we want either both or none, as `val = NULL`  with `val_len > 0` would likely be an error, too.

Reply:
I guess it comes down to the purpose of this assert. The other three asserts notify the developer of an unquestionable runtime problem before a crash. I think this assert should only do the same. I am concerned about prohibiting a developer from doing something when we don't have the context to know if it is a mistake (val && !val_len).

Reply:
changed to `assert(!val_len || (val && val_len))`